subprojects/docs/src/docs/userguide/application_plugin.adoc

```diff
@@ -138,8 +138,12 @@ _Depends on_: `jar`, `startScripts`
 +
 Creates a full distribution TAR archive including runtime libraries and OS specific scripts.
 
+[[sec:application_extension]]
+== Application extension
+
+The application plugin adds an extension to the project, which you can use to configure its behaviour. See the link:{groovyDslPath}/org.gradle.api.plugins.JavaApplication.html[JavaApplication] DSL documentation.
```

Comment:
We need to explain how this relates back to the examples earlier in the chapter. We also need to explain the difference between the extension and the convention properties in terms of syntax

For example, we could say something like the following in this section:

> You can configure the extension via the `application {}` block shown earlier, for example using the following in your build script:
> ```
> application {
>     executableDir = "binaries"
> }
> ```

Reply:
I am a bit bewildered about how to do code blocks in these docs.  They don't seem to be defined inline, at least in this one.  Can they be?

Reply:
Good point. They can be done inline, but we do prefer to have tested samples. We also prefer to have both Groovy and Kotlin samples.

Just use the following syntax and I'll modify to add Kotlin and ensure that it's included in our automated testing:

> ```
> [source,groovy]
> ----
> application {
>     executableDir = "binaries"
> }
> ----
> ```

I could also just update this section myself, but you might find it useful to see the base syntax.